docker-compose.yaml

```diff
@@ -179,6 +181,65 @@ services:
     volumes:
       - ./cdr/cdm/jupyter/user_shared_workspace:/cdm_shared_workspace/user_shared_workspace
 
+  dev_jupyterhub:
+      build:
+        context: .
+        dockerfile: Dockerfile
+      container_name: dev-jupyterhub
+      ports:
+        - "4043:4043"
+      depends_on:
+        - spark-master
+        - minio-create-bucket
+      environment:
+        - NOTEBOOK_PORT=4043
+        - JUPYTER_MODE=jupyterhub
+        - YARN_RESOURCE_MANAGER_URL=http://yarn-resourcemanager:8032
+        - SPARK_MASTER_URL=spark://spark-master:7077
+        - SPARK_DRIVER_HOST=dev-jupterhub
+        - MINIO_URL=http://minio:9002
+        - MINIO_ACCESS_KEY=minio-readwrite
+        - MINIO_SECRET_KEY=minio123
+        - S3_YARN_BUCKET=yarn
+        - MAX_EXECUTORS=4
+        - POSTGRES_USER=hive
+        - POSTGRES_PASSWORD=hivepassword
+        - POSTGRES_DB=hive
+        - POSTGRES_URL=postgres:5432
+        - USAGE_MODE=dev
+      volumes:
+        - ./cdr/cdm/jupyter:/cdm_shared_workspace
+        - ./cdr/cdm/jupyter/home:/home
+
+  user_jupyterhub:
+    build:
+      context: .
+      dockerfile: Dockerfile
+    container_name: user-jupyterhub
+    ports:
+      - "4044:4044"
+    depends_on:
+      - spark-master
+      - minio-create-bucket
+    environment:
+      - NOTEBOOK_PORT=4044
+      - JUPYTER_MODE=jupyterhub
+      - YARN_RESOURCE_MANAGER_URL=http://yarn-resourcemanager:8032
+      - SPARK_MASTER_URL=spark://spark-master:7077
+      - SPARK_DRIVER_HOST=user-jupyterhub
+      - MINIO_URL=http://minio:9002
+      - MINIO_ACCESS_KEY=minio-readonly
+      - MINIO_SECRET_KEY=minio123
+      - S3_YARN_BUCKET=yarn
+      - JUPYTER_MODE=jupyterhub
+      - MAX_EXECUTORS=4
+      - POSTGRES_USER=hive
+      - POSTGRES_PASSWORD=hivepassword
+      - POSTGRES_DB=hive
+      - POSTGRES_URL=postgres:5432
```

Comment:
Not really the point of this PR but it makes me nervous that users have access to these creds...

Reply:
Even that would make me nervous. It means that any user could blow away the hive tables

Reply:
I updated the todo item to 
```
# TODO: create postgres user r/ only read access to the hive tables
```

Reply:
I guess another question is if users and devs are all in the same jupyterhub instance is it possible to have different environments

Reply:
Users won't since they don't have minIO write permission. I am hoping it can be configured. But TBD.

Reply:
Ok, so users = read only, devs = write, essentially.

Reply:
If we could ever get the remote metastore working that might provide some protection as well, not sure